<launch>
    <arg name="checkpoint_path" default="~/PointPillars/pretrained/epoch_160.pth"/> <arg name="config_path" default="~/PointPillars/configs/config.yaml"/> <arg name="lidar_topic" default="/ouster/points"/>
    <arg name="image_topic" default="/oak/rgb/image_raw"/>

    <arg name="lidar_frame_id" default="ouster_lidar"/> <arg name="camera_optical_frame_id" default="oak_rgb_camera_optical_frame"/> <arg name="score_threshold" default="0.45"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="lidar_to_camera_tf_publisher"
          args="-0.030730 -0.386689 -0.671425 0.500339 -0.499616 0.500034 0.499999 $(arg lidar_frame_id) $(arg camera_optical_frame_id)" />

    <node name="pointpillars_ros_node" pkg="pointpillars_ros" type="pointpillars_ros_node.py" output="screen" respawn="false">
        <param name="checkpoint_path" value="$(arg checkpoint_path)"/>
        <param name="config_path" value="$(arg config_path)"/>

        <param name="lidar_topic" value="$(arg lidar_topic)"/>
        <param name="image_topic" value="$(arg image_topic)"/>
        <param name="camera_K" value="[ [684.83331299, 0.0, 573.37109375], [0.0, 684.60968018, 363.70092773], [0.0, 0.0, 1.0] ]" />
        <param name="camera_D" value="[0.0, 0.0, 0.0, 0.0, 0.0]" />

        <param name="target_frame" value="$(arg camera_optical_frame_id)"/>
        <param name="score_threshold" value="$(arg score_threshold)"/>
    </node>

</launch>
